.github/workflows/docker_test.yml

```
@@ -44,7 +44,7 @@ jobs:
           docker build \
             --no-cache \
             --build-arg fastdds_branch=${{ github.event.inputs.fastdds_branch || env.DEFAULT_FASTDDS_BRANCH }} \
-            --build-arg ddsrouter_branch=${{ github.event.inputs.ddsrouter_branch || env.DEFAULT_DDSROUTER_BRANCH }} \
+            --build-arg ddsrouter_branch=${{ github.event.inputs.ddsrouter_branch || GITHUB_HEAD_REF || env.DEFAULT_DDSROUTER_BRANCH }} \
```

It'd be better to use the merge branch

```suggestion
            --build-arg ddsrouter_branch=${{ github.event.inputs.ddsrouter_branch || ${{ github.ref }} || env.DEFAULT_DDSROUTER_BRANCH }} \
```

---

Updated according to GitHub documentation